packages/drivers/odsp-driver/src/odspDocumentStorageManager.ts

```diff
@@ -479,7 +480,7 @@ export class OdspDocumentStorageService implements IDocumentStorageService {
                 // Clear the cache on 401/403/404 on snapshot fetch from network because this means either the user doesn't have permissions
                 // permissions for the file or it was deleted. So the user will again try to fetch from cache on any failure in future.
                 if (errorType === DriverErrorType.authorizationError || errorType === DriverErrorType.fileNotFoundOrAccessDeniedError) {
-                    await this.cache.persistedCache.removeAllEntriesForDocId(this.documentId);
+                    await this.cache.persistedCache.removeEntries(cacheEntry);
```

Comment:
Can you please validate with SPO folks that any blob IDs are unique across 2 versions of file?
I.e. if epoch changed, then new content of the file can't get new blob that would have ID that was already used in a file with old epoch? Cause if it's not true, then we need to delete blobs as well. But if IDs are unique across such transformations, then we can afford to leave blobs in cache and reuse them (if they are still in the file).

Reply:
Yes, let me confirm.